src/track/serato/beatgrid.cpp

```diff
@@ -443,73 +443,57 @@ void SeratoBeatGrid::setBeats(BeatsPointer pBeats,
     auto pBeatsIterator = pBeats->findBeats(0, trackDurationSamples);
 
     // This might be null if the track doesn't contain any beats
-    if (!pBeatsIterator) {
+    if (!pBeatsIterator || !pBeatsIterator->hasNext()) {
         qWarning() << "Serato Beatgrid: No beats available, exporting empty beatgrid!";
         setTerminalMarker(nullptr);
         setNonTerminalMarkers({});
         return;
     }
 
+    const double beatgridFrameOffset = signalInfo.samples2framesFractional(pBeatsIterator->next());
     double currentBeatPositionFrames = 0;
     double previousBeatPositionFrames = 0;
     double previousDeltaFrames = -1;
-    int beatsSinceLastMarker = 0;
     QList<SeratoBeatGridNonTerminalMarkerPointer> nonTerminalMarkers;
+    {
+        const double positionSecs = signalInfo.frames2secsFractional(
```

Comment:
How about introducing utility functions exportFramePositionToSecs()/importFramePositionFromSecs() for this purpose? Thiis would avoid the nested scope and improve readability.

Or maybe a utility class `PositionConverter` that is instantiated with the timing offset to avoid confusing the two double parameters needed for the conversion. Instead of literally passing the timing offset you could then pass an instance of `PositionConverter` to improve type safety.

Reply:
Maybe that could be done later. I tried and struggled to come up with a good name that doesn't obscure what happens behind the scenes and that is rather short at the same time.

Reply:
Ok